src/praisonai/praisonai/cli/features/doctor/checks/runtime_checks.py

```
@@ -59,6 +59,22 @@ def _get_known_runtimes(self) -> Dict[str, RuntimeInfo]:
+        # is both installed AND implemented/registered. This keeps `doctor` in
+        # sync with the registry-backed `--framework` choices so it never
+        # reports a runtime as available when the matching framework cannot run
+        # (e.g. unimplemented AutoGen v0.4 / AG2 entry-point placeholders).
+        def _runtime_usable(framework_name: str, package_name: str) -> bool:
+            try:
+                from praisonai.framework_adapters.registry import get_default_registry
+                registry = get_default_registry()
+                if framework_name in registry.list_names():
+                    return registry.is_available(framework_name)
+            except Exception:
+                pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)